components/camera/rtsp/rtsp.go

```diff
@@ -0,0 +1,240 @@
+// Package rtsp implements an RTSP camera client for RDK
+package rtsp
+
+import (
+	"context"
+	"image"
+	"io"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/aler9/gortsplib/v2"
+	"github.com/aler9/gortsplib/v2/pkg/base"
+	"github.com/aler9/gortsplib/v2/pkg/liberrors"
+	"github.com/aler9/gortsplib/v2/pkg/url"
+	"github.com/edaniels/golog"
+	"github.com/edaniels/gostream"
+	"github.com/pion/rtp"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/utils"
+)
+
+var (
+	model            = resource.NewDefaultModel("rtsp")
+	clientTerminated = liberrors.ErrClientTerminated{}
+)
+
+func init() {
+	registry.RegisterComponent(camera.Subtype, model, registry.Component{
+		Constructor: func(ctx context.Context, _ registry.Dependencies, cfg config.Component, logger golog.Logger) (interface{}, error) {
+			attrs, ok := cfg.ConvertedAttributes.(*Attrs)
+			if !ok {
+				return nil, utils.NewUnexpectedTypeError(attrs, cfg.ConvertedAttributes)
+			}
+			return NewRTSPCamera(ctx, attrs, logger)
+		},
+	})
+
+	config.RegisterComponentAttributeMapConverter(
+		camera.Subtype,
+		model,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var conf Attrs
+			attrs, err := config.TransformAttributeMapToStruct(&conf, attributes)
+			if err != nil {
+				return nil, err
+			}
+			result, ok := attrs.(*Attrs)
+			if !ok {
+				return nil, utils.NewUnexpectedTypeError(result, attrs)
+			}
+			return result, nil
+		},
+		&Attrs{},
+	)
+}
+
+// Attrs are the config attributes for an RTSP camera model.
+type Attrs struct {
+	Address          string                             `json:"rtsp_address"`
+	IntrinsicParams  *transform.PinholeCameraIntrinsics `json:"intrinsic_parameters,omitempty"`
+	DistortionParams *transform.BrownConrady            `json:"distortion_parameters,omitempty"`
```

Comment:
Mostly because if you put in `transform.Distorter` there, there is no clear JSON schema for a Go interface, while one can automatically be generated for a Go struct. In order to create a JSON schema for a Go interface, it has to know about the possible struct that could fulfill it. You can see an example of that  in https://github.com/viamrobotics/rdk/blob/main/spatialmath/orientation_json.go

Because we don't have a `distorter_json`, the field is left as *transform.BrownConrady. Eventually, once it becomes important to actually support different distorters, we can replace this in the cameras. 

Also, intrinsics and distortions are added to every camera model because of the potential of using them to project things into 3D, in which case these two parameters are necessary for the accurate projection.

Reply:
Ohhh, gotcha. Thanks for the explanation!